bytestring_splitter/__init__.py

```diff
@@ -1,75 +1,113 @@
+            second_message_is_int = isinstance(message_parameters[1], int)
 
             if first_message_is_not_int_or_tuple and second_message_is_int:
-                raise TypeError("You can't specify the length of the message as a direct argument to the constructor.  Instead, pass it as the second argument in a tuple (with the class as the first argument)")
+                raise TypeError(
+                    "You can't specify the length of the message as a direct argument to the constructor.  Instead, pass it as the second argument in a tuple (with the class as the first argument)")
 
     def __call__(self, splittable, return_remainder=False, msgpack_remainder=False):
 
         if not self.is_variable_length:
             if not any((return_remainder, msgpack_remainder)) and len(self) != len(splittable):
-                raise ValueError(
+                raise BytestringSplittingError(
                     """Wrong number of bytes to constitute message types {} - 
                     need {}, got {} \n Did you mean to return the remainder?""".format(
                         self.message_types, len(self), len(splittable)))
             if len(self) is not -1 and len(self) > len(splittable):
-                raise ValueError(
+                raise BytestringSplittingError(
                     """Not enough bytes to constitute
                     message types {} - need {}, got {}""".format(self.message_types,
                                                                  len(self),
                                                                  len(splittable)))
         cursor = 0
-        message_objects = []
+        processed_objects = self.processed_objects_container()
 
         for message_type in self.message_types:
-            message_class, message_length, kwargs = self._parse_message_meta(message_type)
+            message_name, message_class, message_length, kwargs = message_type
 
             if message_length is VariableLengthBytestring:
                 # If this message is of variable length, let's get the length
                 # and advance the cursor past the bytes which represent the length.
-                message_length_as_bytes = splittable[cursor:cursor+VARIABLE_HEADER_LENGTH]
+                message_length_as_bytes = splittable[cursor:cursor + VARIABLE_HEADER_LENGTH]
                 message_length = int.from_bytes(message_length_as_bytes, "big")
                 cursor += VARIABLE_HEADER_LENGTH
 
+            if message_length > len(splittable):
+                error_message = "Can't split a message with more bytes than the original splittable.  {} claimed a length of {}"
+                if message_name:
+                    error_message = error_message.format(message_name, message_length)
+                else:
+                    error_message = error_message.format(message_class, message_length)
+                raise BytestringSplittingError(error_message)
```

Comment:
Now I'm starting to think that perhaps you wanted `BytestringSplittingError` to inherit from `ValueError`?

Reply:
I... don't think so.  I think it might be confusing to catch this error when you're excepting for `ValueError`.  It's possible that it's a type (rather than value) problem.